packages/auth/src/core/strategies/passkey.ts

```diff
@@ -0,0 +1,233 @@
+    const userCredential = await UserCredentialImpl._fromIdTokenResponse(
+      authInternal,
+      operationType,
+      finalizeSignInResponse
+    );
+    await auth.updateCurrentUser(userCredential.user);
+    return userCredential;
+  } catch (error) {
+    if (
+      (error as Error).message.includes(
+        'The operation either timed out or was not allowed.'
+      ) &&
+      !manualSignUp
+    ) {
+      // If the user is not signed up, sign them up anonymously
+      const userCredential = await signInAnonymously(authInternal);
+      const user = userCredential.user;
+      return enrollPasskey(user, name);
+    }
+    return Promise.reject(error);
+  }
+}
+
+/**
+ * Links the user account with the given phone number.
+ *
+ * @param user - The user.
+ *
+ * @public
+ */
+export async function enrollPasskey(
+  user: User,
+  name: string
+): Promise<UserCredential> {
+  const userInternal = getModularInstance(user) as UserInternal;
+  const authInternal = _castAuth(userInternal.auth);
+
+  if (name === '') {
+    name = 'Unnamed account (Web)';
+  }
+
+  // Start Passkey Enrollment
+  const idToken = await userInternal.getIdToken();
+  const startEnrollmentRequest: StartPasskeyEnrollmentRequest = {
+    idToken
+  };
+  const startEnrollmentResponse = await startPasskeyEnrollment(
+    authInternal,
+    startEnrollmentRequest
+  );
+
+  // Create the crendential
+  try {
+    const options = getPasskeyCredentialCreationOptions(
+      startEnrollmentResponse,
+      name
+    );
+    const credential = (await navigator.credentials.create({
+      publicKey: options
+    })) as PublicKeyCredential;
+    const idToken = await userInternal.getIdToken();
+    const finalizeEnrollmentRequest: FinalizePasskeyEnrollmentRequest = {
+      idToken,
+      authenticatorRegistrationResponse: publicKeyCredentialToJSON(credential),
+      name,
+      displayName: name
+    };
+    const finalizeEnrollmentResponse: FinalizePasskeyEnrollmentResponse =
+      await finalizePasskeyEnrollment(authInternal, finalizeEnrollmentRequest);
+
+    const operationType = OperationType.LINK;
+    const userCredential = await UserCredentialImpl._fromIdTokenResponse(
+      userInternal.auth,
+      operationType,
+      finalizeEnrollmentResponse
+    );
+    return userCredential;
+  } catch (err) {
+    return Promise.reject(err);
+  }
+}
+
+function convertExcludeCredentials(
+  options:
+    | PublicKeyCredentialCreationOptions
+    | PublicKeyCredentialRequestOptions
+): void {
+  function base64ToBuffer(base64: string): ArrayBuffer {
+    const binaryStr = atob(base64);
+    const len = binaryStr.length;
+    const bytes = new Uint8Array(len);
+    for (let i = 0; i < len; i++) {
+      bytes[i] = binaryStr.charCodeAt(i);
+    }
+    return bytes.buffer;
+  }
+
+  if ('excludeCredentials' in options && options.excludeCredentials) {
+    for (const cred of options.excludeCredentials) {
+      if (typeof cred.id === 'string') {
```

Comment:
Not really. If the id type is, we convert it to ArrayBuffer. If it's already ArrayBuffer, we don't do anything. If it's any other types, it should already throw an error when assign the options.

Reply:
Do you think it is worth assert ing in the else case that type is ArrayBuffer?

Reply:
Maybe not since we don't assert in the non-testing code right?

Reply:
Got it. I think we don't need to assert it here since it's the default type in `PublicKeyCredentialOptions`. Otherwise, we will need to assert in many other places.

Reply:
This is optional from my side, will leave it to you for next steps.